enum.c

```diff
@@ -871,44 +871,138 @@ ary_inject_op(VALUE ary, VALUE init, VALUE op)
+ *
+ *    # Sum of squares, without initial_operand.
+ *    (1..4).inject {|sum, n| sum + n*n }    # => 30
+ *    # Sum of squares, with initial_operand.
+ *    (1..4).inject(2) {|sum, n| sum + n*n } # => 32
+ *
+ *  <b>Operands</b>
+ *
+ *  If argument +initial_operand+ is not given,
+ *  the operands for +inject+ are simply the elements of +self+.
+ *  Example calls and their operands:
+ *
+ *  - <tt>(1..4).inject(:+)</tt>:: <tt>[1, 2, 3, 4]</tt>.
+ *  - <tt>(1...4).inject(:+)</tt>:: <tt>[1, 2, 3]</tt>.
+ *  - <tt>('a'..'d').inject(:+)</tt>:: <tt>['a', 'b', 'c', 'd']</tt>.
+ *  - <tt>('a'...'d').inject(:+)</tt>:: <tt>['a', 'b', 'c']</tt>.
+ *
+ *  Examples with first operand (which is <tt>self.first</tt>) of various types:
+ *
+ *    # Integer.
+ *    (1..4).inject(:+)                # => 10
+ *    # Float.
+ *    [1.0, 2, 3, 4].inject(:+)        # => 10.0
+ *    # Character.
+ *    ('a'..'d').inject(:+)            # => "abcd"
+ *    # Complex.
+ *    [Complex(1, 2), 3, 4].inject(:+) # => (8+2i)
+ *
+ *  If argument +initial_operand+ is given,
+ *  the operands for +inject+ are that value plus the elements of +self+.
+ *  Example calls their operands:
+ *
+ *  - <tt>(1..4).inject(10, :+)</tt>:: <tt>[10, 1, 2, 3, 4]</tt>.
+ *  - <tt>(1...4).inject(10, :+)</tt>:: <tt>[10, 1, 2, 3]</tt>.
+ *  - <tt>('a'..'d').inject('e', :+)</tt>:: <tt>['e', 'a', 'b', 'c', 'd']</tt>.
+ *  - <tt>('a'...'d').inject('e', :+)</tt>:: <tt>['e', 'a', 'b', 'c']</tt>.
+ *
+ *  Examples with +initial_operand+ of various types:
+ *
+ *    # Integer.
+ *    (1..4).inject(2, :+)               # => 12
+ *    # Float.
+ *    (1..4).inject(2.0, :+)             # => 12.0
+ *    # String.
+ *    ('a'..'d').inject('foo', :+)       # => "fooabcd"
+ *    # Array.
+ *    %w[a b c].inject(['x'], :push)     # => ["x", "a", "b", "c"]
+ *    # Complex.
+ *    (1..4).inject(Complex(2, 2), :+)   # => (12+2i)
+ *
+ *  <b>Combination by Given \Method</b>
+ *
+ *  If the method-name argument +symbol+ is given,
+ *  the operands are combined by that method:
+ *
+ *  - The first and second operands are combined.
+ *  - That result is combined with the third operand.
+ *  - That result is combined with the fourth operand.
+ *  - And so on.
+ *
+ *  The return value from +inject+ is the result of the last combination.
+ *
+ *  This call to +inject+ computes the sum of the operands:
+ *
+ *    (1..4).inject(:+) # => 10
+ *
+ *  Examples with various methods:
+ *
+ *    # Integer addition.
+ *    (1..4).inject(:+)                # => 10
+ *    # Integer multiplication.
+ *    (1..4).inject(:*)                # => 24
+ *    # Character range concatenation.
+ *    ('a'..'d').inject('', :+)        # => "abcd"
+ *    # String array concatenation.
+ *    %w[foo bar baz].inject('', :+)   # => "foobarbaz"
+ *    # Hash update.
+ *    h = [{foo: 0, bar: 1}, {baz: 2}, {bat: 3}].inject(:update)
+ *    h # => {:foo=>0, :bar=>1, :baz=>2, :bat=>3}
+ *    # Hash appending.
```

Comment:
# Hash to array conversion.

Reply:
> 
> 
> `# Hash to array conversion.`

# Hash conversion to nested arrays.

Done.